system-parachains/asset-hubs/asset-hub-kusama/primitives/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Module with configuration which reflects AssetHubKusama runtime setup.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use frame_support::weights::Weight;
+use scale_info::TypeInfo;
+use xcm::prelude::*;
+
+pub use bp_xcm_bridge_hub_router::XcmBridgeHubRouterCall;
+
+/// `AssetHubKusama` Runtime `Call` enum.
+///
+/// The enum represents a subset of possible `Call`s we can send to `AssetHubKusama` chain.
```

Comment:
This `pallet-xcm-bridge-hub-router` (aka `ToPolkadotXcmRouter`) is deployed at the sending chain (Kusama AH here) and it is used to connect to the bridged chain consensus (Polkadot). That's why it is called `ToPolkadot`. So the usage of `ToPolkadot` name is imo justified.

Yet there's an ongoing work to add with-Ethereum pallet, so Kusama AH (or Polkadot AH - I'm not sure) will have another instance of this `pallet-xcm-bridge-hub-router`, which will connect it to the Ethereum consensus and will use the `ToEthereumXcmRouter` name (see eg. https://github.com/paritytech/polkadot-sdk/pull/2522/files#diff-a3b656e8095fe815cf8611c8319aff7b2f32482cdbe7c24ac37a2ce45235f32dR941-R942).

This enum is used to encode and dispatch calls **within** the same consensus. It is used by the **source** (Kusama) BH to say that the bridge is congested to the **source** (Kusama) AH. Then the `pallet-xcm-bridge-hub-router` will start increasing the fee, paid for the Kusama -> Polkadot messages. So it needs to use source chain indices: if message route is `Kusama.AH -> Kusama.BH -> Polkadot.BH -> Polkadot.AH`, it is used in `Kusama.AH <> Kusama.BH` communications.

Reply:
OK I think I understand more. However, some things I still don't see w/r/t what needs to be kept in sync. E.g.:

```
/// All entries here (like pretty much in the entire file) must be kept in sync with
/// `AssetHubKusama` `construct_runtime`, so that we maintain SCALE-compatibility.
```

But you are in the AH Kusama runtime, so you can just use `PalletInfoAccess` to get the index. Do you mean that it needs to be in sync with AH Polkadot? Or that AH Polkadot relies on this index not changing? If you're relying on someone not changing an index down the road, perhaps add a test like [here](https://github.com/polkadot-fellows/runtimes/pull/109/commits/90f334bd2eb31055b3178b0acb6bd5133a7c7ec7)?

We're also in progress for moving `runtime-common` into the Fellowship (see https://github.com/polkadot-fellows/runtimes/issues/59). So perhaps declare some `const` pallet indices that runtimes can import (without of course importing the whole runtime).

Reply:
> But you are in the AH Kusama runtime,

That's not true - we are not in the AH Kusama runtime, we are in a separate crate (`bp-asset-hub-kusama`), which is used to declare this call. Then this crate is used:
- in Kusama BH to encode an XCM message that is then sent to the Kusama AH;
- in Kusama AH to test that it is SCALE-compatible with the actual runtime call.

Previously we were declaring crates like that in bridges repo (which are then published to crates.io from `polkadot-sdk`). But it declares some constants (like this call and its weight) and if we need to change it (because e.g. call weight increases), then it'd be hard to do - we need to cross three repos and do a publish, before changes are actually applied. So we decided to go with a crates in this repo. So this PR adds 4 crates:
- `bp-asset-hub-*` to declare those calls and their weights;
- `bp-bridge-hub-*` to declare cost-related constants (there's also other stuff there, but it is unlikely will ever change).